### PR TITLE
Allow nullable boolean in WHERE clauses and support Expr<bool?> combiners

### PR DESCRIPTION
**Pull request by isoos**

### CODE REVIEW

File: typed_sql/lib/src/typed_sql.expr_ext.dart [hunk 1]
```diff
@@ -322,6 +322,64 @@ extension ExpressionNullableBool on Expr<bool?> {
   ///
   /// If this is `NULL`, [isFalse] will evaluate to `FALSE`.
   Expr<bool> isFalse() => isNotDistinctFrom(Literal.false$);
+
+  /// Negate this expression.
+  ///
+  /// This is equivalent to `NOT this` in SQL, where `NOT NULL` is `NULL`.
+  ///
+  /// Also available as `~` operator.
+  Expr<bool?> not() => ExpressionBoolNot<bool?>(this);
```
Comment (jonasfj) [inline typed_sql/lib/src/typed_sql.expr_ext.dart]: This is dangerous and we should decidedly not allow this to be called `.not`.

We could allow it to be called `.notUnlessNull`, but IMO we should omit this.

File: typed_sql/CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,6 @@
+## 0.1.11
+ * Allow nullable boolean in `WHERE` clauses and support for combining `Expr<bool?>` expressions.
```
Comment (jonasfj) [inline typed_sql/CHANGELOG.md]:
```suggestion
 * Allow nullable boolean in `.where(...)` clauses.
 * Support for combining `Expr<bool?>` expressions.
```

File: typed_sql/test/typed_sql/expr/nullable_bool_test.dart [hunk 3]
```diff
@@ -389,6 +389,204 @@ final _cases = [
...
+    expr: toExpr(null as bool?).and(toExpr(null as bool?)),
+    expected: null,
+  ),
+  (
+    name: 'null.and(true)',
+    expr: toExpr(null as bool?).and(toExpr(true as bool?)),
+    expected: null,
+  ),
+  (
+    name: 'null.and(false)',
```
Comment (jonasfj) [inline typed_sql/test/typed_sql/expr/nullable_bool_test.dart:420]: Also worth testing `toExpr(null).asBool().and(...`

File: typed_sql/test/typed_sql/where/nullable_bool/nullable_bool_where_test.dart
Comment (jonasfj) [inline typed_sql/test/typed_sql/where/nullable_bool/nullable_bool_where_test.dart:1]: Please put this in a folder `typed_sql/test/typed_sql/expr/nullable_bool_where/nullable_bool_where_test.dart`

or even better:
`typed_sql/test/typed_sql/where/nullable_bool/nullable_bool_test.dart`

We probably should be writing a bunch of tests exploring differnet `.where` logic hacks and have such tests live in `typed_sql/test/typed_sql/where/`.

File: typed_sql/test/typed_sql/where/nullable_bool/nullable_bool_where_test.dart [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+  Future<List<int>> idsWhere(
+    Database<TestDatabase> db,
+    Expr<bool?> Function(Expr<Item> item) conditionBuilder,
+  ) async {
+    final ids = await db.items
+        .where(conditionBuilder)
+        .select((item) => (item.id,))
+        .fetch();
+    return ids.toList()..sort();
+  }
```
Comment (jonasfj) [inline typed_sql/test/typed_sql/where/nullable_bool/nullable_bool_where_test.dart:114]:
```suggestion
  final _cases = <(String name, Expr<bool?> Function(Expr<Item> item) where, List<int> expected)>[
    (
      name: '.where(.value)',
      where: (item) => item.value,
      expected: [1],
    ),
  ];
```

it'll be easier to read, than these special functions.

**Review by jonasfj — CHANGES_REQUESTED** [review]

**Review by jonasfj — APPROVED** [review]
